cli_tools/gce_vm_image_import/main.go

```diff
@@ -18,119 +18,117 @@ package main
-		project, *scratchBucketGcsPath, *oauth, *ce, *gcsLogsDisabled, *cloudLogsDisabled,
-		*stdoutLogsDisabled, *noExternalIP, *labels, currentExecutablePath, *storageLocation,
-		*uefiCompatible, *sysprepWindows, storageClient, paramPopulator)
-	return service.NewLoggableFromWorkflow(wf), err
+	// 3. Parse, validate, and populate arguments.
+	return args.ParseArgs(os.Args[1:], paramPopulator, sourceFactory)
 }
 
-func main() {
-	flag.Parse()
+// terminate is used when there is a failure prior to running import. It sends
+// a message to the logging framework, and then executes os.Exit(1).
+func terminate(args args.ImporterArguments, cause error) {
+	noOpCallback := func() (service.Loggable, error) {
+		return nil, cause
+	}
+	// Ignoring the returned error since its a copy of
+	// the return value from the callback.
+	_ = service.RunWithServerLogging(service.ImageImportAction, initLoggingParams(args), nil, noOpCallback)
+	os.Exit(1)
+}
 
-	paramLog := service.InputParams{
+func initLoggingParams(args args.ImporterArguments) service.InputParams {
+	return service.InputParams{
 		ImageImportParams: &service.ImageImportParams{
 			CommonParams: &service.CommonParams{
-				ClientID:                *clientID,
-				Network:                 *network,
-				Subnet:                  *subnet,
-				Zone:                    *zone,
-				Timeout:                 *timeout,
-				Project:                 *project,
-				ObfuscatedProject:       service.Hash(*project),
-				Labels:                  *labels,
-				ScratchBucketGcsPath:    *scratchBucketGcsPath,
-				Oauth:                   *oauth,
-				ComputeEndpointOverride: *ce,
-				DisableGcsLogging:       *gcsLogsDisabled,
-				DisableCloudLogging:     *cloudLogsDisabled,
-				DisableStdoutLogging:    *stdoutLogsDisabled,
+				ClientID:                args.Environment.ClientID,
+				Network:                 args.Environment.Network,
+				Subnet:                  args.Environment.Subnet,
+				Zone:                    args.Environment.Zone,
+				Timeout:                 args.Translation.Timeout.String(),
+				Project:                 args.Environment.Project,
+				ObfuscatedProject:       service.Hash(args.Environment.Project),
+				Labels:                  fmt.Sprintf("%v", args.Image.Labels),
+				ScratchBucketGcsPath:    args.Environment.ScratchBucketGcsPath,
+				Oauth:                   args.Environment.Oauth,
+				ComputeEndpointOverride: args.Environment.ComputeEndpoint,
+				DisableGcsLogging:       args.Environment.GcsLogsDisabled,
+				DisableCloudLogging:     args.Environment.CloudLogsDisabled,
+				DisableStdoutLogging:    args.Environment.StdoutLogsDisabled,
 			},
-			ImageName:          *imageName,
-			DataDisk:           *dataDisk,
-			OS:                 *osID,
-			SourceFile:         *sourceFile,
-			SourceImage:        *sourceImage,
-			NoGuestEnvironment: *noGuestEnvironment,
-			Family:             *family,
-			Description:        *description,
```

Comment:
Description missing is a bug?

Reply:
Ugh. Yeah. I did a hard reset on this to make sure it was fully intact.